lib/Doctrine/ORM/Mapping/ClassMetadata.php

```diff
@@ -1002,6 +1003,11 @@ public function addDiscriminatorMapClass($name, string $className) : void
             throw MappingException::invalidClassInDiscriminatorMap($className, $this->className);
         }
 
+        $refl = new ReflectionClass($className);
+        if ($refl->name !== $className) {
+            throw MappingException::invalidClassInDiscriminatorMap($className, $this->className);
+        }
```

Comment:
I'm opened to suggestion if you see a more efficient way to grab actual class name without using reflection.

Reply:
However, I'm not sure efficiency is really researched here, as metadata are cached.

Reply:
I don't think the reflection API performance are that bad, and since this is about metadata, there is probably caching after this, right? So it's not a hot path, and it's cached, isn't it?

Reply:
That's my belief as well :+1: